src/advanced/cancel_contracts.md

```diff
@@ -0,0 +1,31 @@
+# Canceling contracts
+
+Right now there is no easy, friendly way to cancel all of your contracts (it's planned to have an easier way)
+
+from the graphql service execute the following query.
+
+```
+query MyQuery {
+
+  nodeContracts(where: {twinId_eq: TWIN_ID, state_eq: Created}) {
+    contractId
+  }
+}
+
+```
+
+replace `TWIN_ID` with your twin id. the information should be available on the [Dashboard](/dashboard/)
+
+then from [polkadot UI](https://polkadot.js.org/apps/), add the tfchain endpoint to development
+
+![](img/polka_web_add_development_url.png)
+
+go to extrinsics, choose the smartContract module and cancelContract extrinsic and use the IDs from graphql to execute the cancelation
+
+![](img/polka_web_cancel_contracts.jpg)
+
+## cancel using code
+
+can also cancel them using [code](https://github.com/threefoldtech/grid3_client_ts/blob/development/scripts/delete_all_contracts.ts)
+
+after cloning `grid3_client_ts`, `yarn install`, and update the `config.json` (beside the script), execute  `yarn run ts-node --project ./tsconfig-node.json scripts/delete_all_contracts.ts`
```

Comment:
This section is still outdated:

The link should be replaced since it points to the deprecated grid3_client_ts repo. should be replaced with [this link](https://github.com/threefoldtech/tfgrid-sdk-ts/tree/development/packages/grid_client).

and the text under it should also be updated since now you will need to clone and build [tf-grid-sdk-ts](https://github.com/threefoldtech/tfgrid-sdk-ts) instead of `grid3_client_ts`